lib/internal/cluster/master.js

```diff
@@ -104,8 +104,21 @@ function createWorkerProcess(id, env) {
   workerEnv.NODE_UNIQUE_ID = '' + id;
 
   if (execArgv.some((arg) => arg.match(debugArgRegex))) {
-    execArgv.push(`--inspect-port=${process.debugPort + debugPortOffset}`);
-    debugPortOffset++;
+    let inspectPort;
+    if ('inspectPort' in cluster.settings) {
+      if (typeof cluster.settings.inspectPort === 'function')
+        inspectPort = cluster.settings.inspectPort();
```

Comment:
Done.

Test file became less readable, is it okay? I think I can refactor it, but is it really worth it?